subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpResponseResource.java

```diff
@@ -68,6 +74,27 @@ public int getStatusCode() {
         return response.getStatusLine().getStatusCode();
     }
 
+    public Date getValidUntil() {
+        Header cacheControlHeader = response.getFirstHeader(HttpHeaders.CACHE_CONTROL);
+        Header dateHeader = response.getFirstHeader(HttpHeaders.DATE);
```

Comment:
We should be calculating the expiry date based on the local time, not the server time.